pandas/core/arrays/boolean.py

```diff
@@ -320,7 +320,8 @@ def _values_for_factorize(self) -> Tuple[np.ndarray, int]:
 
     @classmethod
     def _from_factorized(cls, values, original: "BooleanArray") -> "BooleanArray":
-        return cls._from_sequence(values, dtype=original.dtype)
+        mask = values == -1
+        return cls(values.astype(bool), mask)
```

Comment:
The `astype` causes a copy, which I think wasn't present before. This could perhaps be a `.view('bool')` to preserve the previous no-copy behavior.

Reply:
changed to `.astype(bool, copy=False)`